airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/commands.py

```diff
@@ -99,6 +99,12 @@ def crash_on_any_failure(poetry_package_poe_tasks_results: List[Tuple[Path, asyn
     type=click.Choice(INTERNAL_POETRY_PACKAGES),
     multiple=True,
 )
+@click.option(
```

Comment:
I don't think the introduction of this option is necessary. If users wants to run tests locally on different python version I'd prefer them to either modify their pyproject.toml or switch environments via poetry.

Reply:
@alafanechere but then it's not much of a matrix if it's a single point. Also, poe_tasks don't run concurrently across versions. I intentionally moved the task group inside the versions iterator.

The argument in question is dedicated to CI. If we define (or import) it as a matrix on Github Actions, we start independent containers to run every version. I don't think the main benefit is independent parallelism. It's more about handling them as independently addressable cases of the workflow that you can assign different rules to. And they are handled separately in the UI so it's very transparent. That's important imho if we want to test a range of versions.

Reply:
> It's more about handling them as independently addressable cases of the workflow that you can assign different rules to. And they are handled separately in the UI so it's very transparent. That's important imho if we want to test a range of versions.

I don't want `airbyte-ci` or GHA worfklow to enforce which python version a package should support. So this notion of higher level of control via GHA matrixes is not something I'd push for. 
I'm not 💯 sure this is what your meant though.